src/lib_json/json_reader.cpp

```diff
@@ -1587,8 +1582,8 @@ bool OurReader::decodeNumber(Token& token, Value& decoded) {
   if (isNegative) {
     // We use the same magnitude assumption here, just in case.
     const Value::UInt last_digit = value % 10;
-    decoded = -Value::LargestInt(value / 10) - last_digit;
-  } else if (value > Value::LargestUInt(Value::maxLargestInt)) {
+    decoded = -Value::LargestInt(value / 10) * 10 - last_digit;
+  } else if (value <= Value::LargestUInt(Value::maxLargestInt)) {
```

Comment:
Oh! This was a bug? Good catch.

Reply:
Just wonder the bug, is there a example code to reproduce the bug?